### PR TITLE
feat: add clarinet tool and first tests

### PR DESCRIPTION
**Pull request by LNow**
This PR introduces  [**clarinet**](https://github.com/lgalabru/clarinet) to the project, reorganize directory structure and adds npm's `package.json` file to provide more user friendly commands.

Unfortunately right now clarinet cannot simulate the deployment of a contract by any arbitrary address so if we write our contract so that they can be deployed into mainnet without any modifications it will throw `Analysis error: use of undeclared trait <ft-trait>` error as it is not aware of any contract available on the mainnet.

To mitigate this issue I decided that we must have a mechanism, that will alter our contracts and replace all mainnet addresses used in `impl-trait` with address known to clarinet.

That's how I created a small bash script `clarinet-prepare.sh`.  This is a script we should execute before calling clarinet command.
In addition to that to make testing certain features much easier, I came up with an idea of **contract test add-ons** which are small Clarity code snippets that `clarinet-prepare.sh` appends at the end of contract files.  
These add-ons can be used to expose functions which are defined in contract as private, or allow us perform operations such as minting tokens bypassing complex contract logic.

`clarinet-prepare.sh` executes following steps.
1. copies all contract files from `./contracts` to `./contracts/clarinet`
2. modify address used in `impl-trait` by replacing them with different address. ie: 
`(impl-trait 'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-10-ft-standard.ft-trait)` 
is converted to this: `(impl-trait 'ST1HTBVD3JG9C05J7HBJTHGR0GGW7KXW28M5JS8QE.sip-10-ft-standard.ft-trait)`
3. appends at the end of file content of test add-on file with the same name, stored in `./contracts/test_addons`

**New commands**
- `npm run clarinet:prepare`
- `npm run clarinet:test`, `npm test`, `npm run test`
- `npm run clarinet:check`
- `npm run clarinet:console`
- `npm test:native`

`clarinet:test`, `clarinet:check` and `clarinet:console` executes in background `clarinet:prepare` so we don't have to call it every single time.

Old tests prepared by @jcnelson can still be executed using:
```sh
$ ./tests/run-tests.sh 
```

### CODE REVIEW

File: src/citycoin-client.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  }
+
+  /**
+   * Mints token to make testing easier.
+   * 
+   * @param amount 
+   * @param recipient 
+   */
+  ftMint(amount: number, recipient: Account) {
+    const block = this.chain.mineBlock([
```
Comment (friedger) [inline src/citycoin-client.ts:38]: should be without `mineBlock` wrap

**Review by friedger — APPROVED** [review]
Looks good

File: src/citycoin-client.ts (hunk 1 above)
Comment (LNow) [reply src/citycoin-client.ts]: You're right. I corrected it with  6a6f727

**Review by pxydn — APPROVED** [review]
LGTM

**Review by pxydn — APPROVED** [review]

**Review by whoabuddy — APPROVED** [review]
LGTM - thank you!

**Pull request merged**